core/src/main/java/org/apache/gravitino/audit/AuditLogWriter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.audit;
+
+import java.util.Map;
+import org.apache.gravitino.listener.api.event.Event;
+
+/**
+ * Interface for writing the audit log, which can write to different storage, such as file,
+ * database,mq.
+ */
+public interface AuditLogWriter {
+
+  /** @return formatter. */
+  Formatter getFormatter();
+
+  /**
+   * Initialize the writer with the given configuration.
+   *
+   * @param properties
+   */
+  void init(Formatter formatter, Map<String, String> properties);
```

Comment:
We can have properties for different writer, so my suggestion is that we can do one more interface to represent writer name like:

```
String name();
```

For example file writer can be "file". Then this specific writer related configurations can be "gravitino.audit.writer.file.xxx", what do you think?